packages/connectivity/src/scp-cf/destination/destination-service.ts

```diff
@@ -209,11 +257,45 @@ function errorMessageFromResponse(
     : '';
 }
 
-function callDestinationService(
+async function callCertificateEndpoint(
+  uri: string,
+  headers: Record<string, any>,
+  options?: ResilienceOptions
+): Promise<AxiosResponse<DestinationCertificate>> {
+  if (!uri.includes('Certificates')) {
+    throw new Error(
+      `callCertificateEndpoint was called with illegal arrgument: ${uri}. URL must be certificate endpoint of destination service.`
+    );
+  }
+  return callDestinationService(uri, headers, options) as Promise<
+    AxiosResponse<DestinationCertificate>
+  >;
+}
+
+async function callDestinationEndpoint(
   uri: string,
   headers: Record<string, any>,
   options?: ResilienceOptions
 ): Promise<AxiosResponse<DestinationJson | DestinationConfiguration>> {
+  if (!uri.match(/v1\/.*[d,D]estinations/)) {
```

Comment:
There is a code scanning alert for this line, can you check this?

Reply:
made the regex less fuzzy improving performance